cowsay: refresh (#5934)
tldr-pages/tldr: pages/common/cowsay.md

# cowsay

> > Print an ASCII friend (by default a cow) saying or thinking something.

Reviewer: Why `friend` instead of `character`?

Author: since later on I have used "character" to signify letters rather than the ASCII friends.

Also the -f to me means friend, which is a little easier to remember for users rather than "cow-file"

Reviewer: yeah, the source code refers to those as `cowfile` and they are inside the https://github.com/tnalpgge/rank-amateur-cowsay/tree/master/cows folder. :cow:

Reviewer 2: I don't particularly mind this change.  "character" (_especially_ when combined with "ASCII" 🙂) frequently is thought of as a text character, such as 'A' or '>'

Reviewer 2: A more maintained version of the source is here: https://github.com/cowsay-org/cowsay

Author: 2>1, should I resolve this?

Reviewer 2: @navarroaxel your opinion?

Reviewer: Actually I don't like friend, I prefer character.

Reviewer 2: Perhaps we could use `animal` or `creature`?

Reviewer 2: I wonder if @sbrl has any opinions?

Reviewer 3: What about `artwork`?

Reviewer: `friend` sound to interaction, `artwork` static, I like it.

Author: I feel art is better instead of artwork

Author: I will change this now
I will use art.

Author: We cannot use animal or creature since some things are animals/creatures E.G. this block of cheese

```
 ________________________
< You can't use animals! >
 ------------------------
   \
    \
      _____   _________
     /     \_/         |
    |                 ||
    |                 ||
   |    ###\  /###   | |
   |     0  \/  0    | |
  /|                 | |
 / |        <        |\ \
| /|                 | | |
| |     \_______/   |  | |
| |                 | / /
/||                 /|||
   ----------------|
        | |    | |
        ***    ***
       /___\  /___\
```

Author: Cowsay, isn't meant to be a serious program, its not like dd(disk destroyer), its light-hearted, and meant to just be used for fun, if someone is using cowsay, they are having fun or just bored. They aren't probably going to care if they see friend, rather than artwork. When a program is stored in /usr/bin/games, you know its a game/joke program. I don't see why it has to be so serious 

However axel does have a point point about it being interactive using the word friend, I see where he is coming from, however just from the description, someone can work out that cowsay isn't meant to be a robot friend or whatever

Author: What are we going to use?

Reviewer 2: You raise good points, @CleanMachine1.  Why not go with `art`?

Author: Look at this, the first line 

> Print an ASCII friend

How can you change that to art without making it sound stupid

Reviewer 2: Remove the article.

>Print ASCII art.

Author: Yes thats the answer, pretty obvious but it makes the rest sound wrong as well

> Print ASCII art (by default a cow) saying or thinking something.

Friend fits that more

Reviewer 2: I don't think it sounds that bad, actually.  If cheese can talk, so can art 🙂

Author: Make proposed commit and then we will use art. I would like to get rid of this branch since its the last of my fork meaning I can delete it and clone the upstream

Reviewer 2: ```suggestion
> Print ASCII art (by default a cow) saying or thinking something.
```

Reviewer 2: Done, @CleanMachine1.

> More information: <https://github.com/tnalpgge/rank-amateur-cowsay>.

- Print an ASCII cow saying "Hello World":

> `cowsay "Hello World"`

Reviewer 2: ```suggestion
`cowsay "{{Hello World}}"`
```

> `cowsay "{{Hello World}}"`

Reviewer: I dont like the title case in this sentence.

Author: What would you have it as then?
Isn't the top usage meant to be the most basic, which will be built off from to have more advanced commands.

Reviewer 2: I'm not sure how much I like it either, now that you bring it up.  However, it seems to be more common across the pages (32 vs. 13) 🤔

Reviewer 2: Probably `Hello world`.

Author: Deleted my post since it was irrelevant to what was being asked...

Well I feel if we are going to change this one, then the rest need to be changed

Author: I personally believe it should be one of the following

> Hello World
hello world
Hello, World
Hello, World!

Reviewer 2: I like the last one the best.

Reviewer 3: I suggest the following grammar:
```suggestion
`cowsay "{{hello, world}}"`
```

The reasoning for this is that the [first known instance of a hello world](https://en.wikipedia.org/wiki/%22Hello,_World!%22_program#History) uses this exact casing and grammar.

Author: Should I apply that throughout?

> - Read text from stdin for the balloon:

Reviewer 2: ```suggestion
- Print an ASCII cow saying text from stdin:
```

Author: This comment was left from the original page, however it sounds a little man pagey, so i'll commit that

> `echo "Hello World" | cowsay`

Reviewer 2: ```suggestion
`echo "{{Hello World}}" | cowsay`
```

> - List all available friends:

Reviewer 2: I like to use "valid" for list commands like this, because it clarifies it a bit more than "available" 🤔
```suggestion
- List all valid friends:
```

Author: I feel valid implies that there are situations when you cant use the friends; available does as well but less so

Reviewer 2: Maybe you're right — let's skip this.

> - List all available friends:

Author: What about this?

Author: When it comes to the friend,

`cowsay -l`

> - Print an ASCII friend (different from default cow) saying "Hello World":

Reviewer 2: ```suggestion
- Print a custom ASCII friend saying "Hello World":
```
I appreciate the correct preposition ("from") after "different"! 😉

Author: I am unsure about this one, I feel since this is a fun "game" program, it should be a bit lighter and less serious than that.

Reviewer 2: To me, they both seem equally (un)serious, and the latter is a bit easier to read 🤔

`cowsay -f {{friend}} "Hello World"`

> - Print a stoned thinking ASCII cow:

Reviewer 2: How about we pick something else?  "stoned" is not very obvious.
```suggestion
- Print a dead thinking ASCII cow:
```

Author: I didn't use the word stoned, the original maker did, i'll change that happily.

> `cowsay -f {{friend}} "{{hello, world}}"`

Reviewer 2: ```suggestion
`cowsay -f {{art}} "{{hello, world}}"`
```

> `cowthink -s "I'm just a cow, not a great thinker..."`

Reviewer 2: ```suggestion
`cowthink -s "{{I'm just a cow, not a great thinker...}}"`
```

> `cowthink -s "{{I'm just a cow, not a great thinker...}}"`

Reviewer 2: ```suggestion
`cowthink -d "{{I'm just a cow, not a great thinker...}}"`
```

> - Print a ASCII cow with custom eyes:

Reviewer 2: ```suggestion
- Print an ASCII cow with custom eyes:
```

Author: good spot
